### PR TITLE
feat: adding an attachedTrigger with sheet overlay

### PR DESCRIPTION
**Pull request by anandtiwary**
## Description
Please include a summary of the change, motivation and context.
feat: adding an attachedTrigger with sheet overlay

### Testing
Please describe the tests that you ran to verify your changes. Please summarize what did you test and what needs to be tested e.g. deployed and tested helm chart locally. 

### Checklist:
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been merged and published in downstream modules

### Documentation
Make sure that you have documented corresponding changes in this repository or [hypertrace docs repo](https://github.com/hypertrace/hypertrace-docs-website) if required.

### CODE REVIEW

File: projects/components/src/overlay/sheet/sheet-overlay.component.scss [hunk 1]
```diff
@@ -62,3 +43,17 @@
     }
   }
 }
+
+.attached-trigger {
+  cursor: pointer;
+  background: white;
+  width: 200px;
```
Comment (itssharmasandeep) [inline projects/components/src/overlay/sheet/sheet-overlay.component.scss:50]: Should we avoid this? I know it works for our current use case.

File: projects/components/src/overlay/sheet/sheet-overlay.component.scss [hunk 2]
```diff
@@ -62,3 +43,17 @@
...
+  height: 40px;
+  transform: translate(calc(-50% - 20px), 0px) rotate(270deg);
+  position: relative;
+  top: -50%;
+  border-radius: 4px;
+  display: flex;
+  align-items: center;
+  justify-content: center;
```
Comment (itssharmasandeep) [inline projects/components/src/overlay/sheet/sheet-overlay.component.scss:58]:
```
box-shadow: -3px 1px 24px rgba(0, 0, 0, 0.12), -1px -1px 8px rgba(0, 0, 0, 0.08);
clip-path: inset(-20px -20px 0px -20px);
```
We should also add these two properties here as well to have proper box-shadow in non collapsed view. You can try this.

File: projects/components/src/overlay/sheet/sheet-overlay.component.scss [hunk 3]
```diff
@@ -62,3 +43,17 @@
...
+.attached-trigger {
+  cursor: pointer;
+  background: white;
+  width: 200px;
+  height: 40px;
+  transform: translate(calc(-50% - 20px), 0px) rotate(270deg);
+  position: relative;
+  top: -50%;
+  border-radius: 4px;
+  display: flex;
```
Comment (itssharmasandeep) [inline projects/components/src/overlay/sheet/sheet-overlay.component.scss:56]: nit: `@include center-contents`

File: projects/components/src/overlay/sheet/sheet-overlay.component.scss (hunk 1 above)
Comment (anandtiwary) [reply projects/components/src/overlay/sheet/sheet-overlay.component.scss:50]: Let's keep this for now. I want to put some restrictions on the content.

File: projects/components/src/overlay/sheet/sheet-overlay.component.scss (hunk 3 above)
Comment (anandtiwary) [reply projects/components/src/overlay/sheet/sheet-overlay.component.scss:56]: I prefer putting flex properties directly :)

File: projects/components/src/overlay/sheet/sheet-overlay.component.ts [hunk 4]
```diff
@@ -12,29 +12,36 @@ import { SheetOverlayConfig, SheetSize } from './sheet';
...
-      <div class="content-wrapper">
-        <div class="content">
-          <ng-container *ngIf="this.isComponentSheet; else templateRenderer">
-            <ng-container *ngComponentOutlet="this.renderer; injector: this.rendererInjector"></ng-container>
-          </ng-container>
-          <ng-template #templateRenderer>
-            <ng-container *ngTemplateOutlet="this.renderer"></ng-container>
-          </ng-template>
-        </div>
+      <div class="attached-trigger" *ngIf="!!this.attachedTriggerTemplate" (click)="this.toggleCollapseExpand()">
```
[the remark this answers is not in the document]
Comment (anandtiwary) [reply projects/components/src/overlay/sheet/sheet-overlay.component.ts:41]: added

File: projects/components/src/overlay/sheet/sheet-overlay.component.scss (hunk 2 above)
Comment (anandtiwary) [reply projects/components/src/overlay/sheet/sheet-overlay.component.scss:58]: added

File: projects/components/src/overlay/sheet/sheet-overlay.component.scss [hunk 5]
```diff
@@ -13,26 +16,6 @@
   flex-direction: column;
   z-index: 1;
 
-  &.sheet-size-responsive-extra-large {
```
Comment (palbizu) [inline projects/components/src/overlay/sheet/sheet-overlay.component.scss:16]: Why this size is not necessary anymore?

Comment (anandtiwary) [reply projects/components/src/overlay/sheet/sheet-overlay.component.scss:16]: Adding it programmatically in TS

**Review by palbizu — APPROVED** [review]
LGTM!

File: projects/components/src/overlay/sheet/sheet-overlay.component.ts [hunk 6]
```diff
@@ -46,6 +60,8 @@ export class SheetOverlayComponent {
   public readonly rendererInjector: Injector;
   public visible: boolean = true;
   public readonly closeOnEscape: boolean;
+  public readonly attachedTriggerTemplate?: TemplateRef<unknown>;
+  public isViewCollapsed: boolean;
```
Comment (arjunlalb) [inline projects/components/src/overlay/sheet/sheet-overlay.component.ts:64]: can we repurpose the `visible` property for this? both seems to be doing similar things.

Comment (anandtiwary) [reply projects/components/src/overlay/sheet/sheet-overlay.component.ts:64]: Visible is actually hiding the entire content (including the trigger) before the popover is closed. So, it is different than `isViewCollapsed` and I can't combine them. Also, keeping them separate keeps the blast radius small

**Pull request merged**